File: springboot-backend/src/main/java/com/capstone/project/service/FeedbackService.java
package com.capstone.project.service;

import com.capstone.project.exception.ResourceNotFroundException;
import com.capstone.project.model.Feedback;

import java.util.List;

public interface FeedbackService {

    List<Feedback> getAllFeedbacks();

    Feedback createFeedback(Feedback feedback);

    Feedback getFeedbackById(int id) throws ResourceNotFroundException;

    Feedback updateFeedback(int id, Feedback feedbackDetails) throws ResourceNotFroundException;

    Boolean deleteFeedback(int id) throws ResourceNotFroundException;
}

File: react-frontend/src/services/FeedbackService.js
import axios from "axios";

const API_BASE_URL = "http://localhost:8080/api/v1";

const createFeedback = (feedback) => {
  return axios.post(API_BASE_URL + "/feedbacks", feedback);
};

const getFeedbackById = (id) => {
  return axios.get(API_BASE_URL + "/feedbacks/" + id);
};

const FeedbackService = {
  createFeedback,
  getFeedbackById,
};

export default FeedbackService;
